### DISCUSSION TITLE
Should we keep drone?

### DISCUSSION THREAD

**Discussion by aucampia** [RDFLib/rdflib#1669] [Development]
Should we keep drone running? I guess it does not hurt, but I'm not sure it helps that much either, probably something we should think about.

CC: @ashleysommer @nicholascar

**Comment by nicholascar** [comment]
This is definitely one for @ashleysommer who set drone up! It was a replacement for Travis which was going to start to cost , but if we’ve emulated Drone’s actions without it, we should remove it as we have a lot of moving parts in the RDFlib testing system so would be good to simplify.

**Comment by ashleysommer** [comment]
I didn't know we added GitHub Actions. What was the reasoning behind that? When we had to move from Travis, we discussed alternatives and decided on Drone because it was very close in features and functionality to Travis, and we could host it ourselves, and have more control over it. We were worried about vendor lockin with GitHub-specific tooling.

If we have really decided to go back on that and go with GitHub Actions, then we can probably retire our drone server.

**Reply by aucampia** [reply]
I think vendor lockin is a valid concern, but the way that github actions workflow is set up (see [this](https://github.com/RDFLib/rdflib/blob/master/.github/workflows/validate.yaml)) and are evolving (see [this](https://github.com/RDFLib/rdflib/pull/1672/files#diff-9237aa56078dfbd8fcac3198563f306ada599aabaa43aca884905129779c5846)) is fairly decoupled from github actions. And I'm trying to more heavily lean on tox so that contributors can be confident that if tox passes github actions pass.

The benefit of github actions are that we can run the workflow on windows and macos, and there were quite a few problems on windows and even macos, that snuck in over time, so it is worth validating on those.

**Reply by nicholascar** [reply]
> I didn't know we added GitHub Actions. What was the reasoning behind that?

I guess contributors - @aucampia and others - know how to use them well!

> If we have really decided to go back on that and go with GitHub Actions...

We may have fallen into a decision, rather than making a positive choice.

So we do need to re-address our overall situation. Perhaps we need a meeting of all 4 maintainers.

**Reply by aucampia** [reply]
Any update on this? I made https://github.com/RDFLib/rdflib/issues/1815 - this is basically what I would say is the only remaining thing stopping us from disabling drone.

**Reply by nicholascar** [reply]
Let's schedule a quick maintainers call. I think we all have good intentions to be in sych here but we are missing thing by not actually communicating synchronously! 

How does a virtual meeting on Friday 22nd April, a week from now, at 10:00 UTC (https://www.timeanddate.com/worldclock/converter.html?iso=20220422T100000&p1=47&p2=1440&p3=37) sound? We need only meet for a short time but we can just tick off a few things we should all agree on, like this, potential release timing etc.

**Reply by aucampia** [reply]
Time works for me.

**Reply by aucampia** [reply]
Any time other than 01h00 -> 07h00  CET (09h00 -> 15h00 AEST) is really fine by me, just depends on whether I have something on my schedule already.

**Reply by nicholascar** [reply]
OK @white-gecko and @ashleysommer , does this time work for you?
